src/lib.rs

```diff
@@ -605,6 +608,60 @@ macro_rules! query_group {
                     db_trait($query_trait);
                     query_type($QueryType);
                     key($($key_name: $key_ty),*);
+                    lifetime();
+                ]
+            }
+        )*
+    };
+
+    // Base case with lifetime: found the trait body
+    (
+        attr[$($trait_attr:tt)*];
+        headers[$v:vis, $query_trait:ident, $($header:tt)*];
+        tokens[{
+            $(
+                $(#[$method_attr:meta])*
+                fn $method_name:ident($($key_name:ident: $key_ty:ty),* $(,)*) -> $value_ty:ty {
+                    type $QueryType:ident;
+                    $(storage $storage:tt;)* // FIXME(rust-lang/rust#48075) should be `?`
+                    $(use fn $fn_path:path;)* // FIXME(rust-lang/rust#48075) should be `?`
+                }
+            )*
+        }];
+        lifetime[$lt:lifetime];
+    ) => {
+        $($trait_attr)* $v trait $query_trait<$lt>: $($crate::plumbing::GetQueryTable<$QueryType<$lt>> +)* $($header)* {
+            $(
+                $(#[$method_attr])*
+                fn $method_name(&self, $($key_name: $key_ty),*) -> $value_ty {
+                    <Self as $crate::plumbing::GetQueryTable<$QueryType<$lt>>>::get_query_table(self)
+                        .get(($($key_name),*))
+                }
+            )*
+        }
+
+        $(
+            #[derive(Default, Debug)]
+            $v struct $QueryType<$lt>(std::marker::PhantomData<&$lt ()>);
```

Comment:
Similarly here I would use a phantomdata like:

```
PhantomData<($(&$lt (),)*)>
```

which then also works regardless of how many lifetimes were provided.

*But* this is going to be a drag: some of the APIs really rely on being able to type `Query` as a value and have it work. Hmm.

Good reason to push harder on https://github.com/rust-lang/rfcs/pull/2357 =)

Reply:
Or, alternatively, perhaps a reason to rejigger some of those APIs..

Reply:
e.g., `db.query_mut(Foo).set(...)`